app/bundles/CoreBundle/Doctrine/Common/DataFixtures/Event/PreExecuteEvent.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Doctrine\Common\DataFixtures\Event;
+
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class PreExecuteEvent extends Event
+{
+    private EntityManagerInterface $entityManager;
+    private int $purgeMode;
+
+    public function __construct(EntityManagerInterface $entityManager, int $purgeMode)
```

Comment:
I'm not a big fan of adding services to the events. We have that in some events and it's hard to maintain. The subscriber that is listening for this event should have the EntityManager as a dependency.

Reply:
@escopecz I tend to do the same. EntityManager is in the event, because DI doesn't know which is passed here: https://github.com/mautic/mautic/pull/11701/files#diff-cbc1d88e43db02bb48a776328bd26d51e8e5549ba0deac052013d9d0344ed088R31

Reply:
I still don't get it. Passing Entity Manager [here](https://github.com/mautic/mautic/pull/11701/files#diff-c6fbb3dc5b1b873b78373a48784cbf8f1f81954bc517b3dfe8b93784300836ecR53) wouldn't work?

Reply:
No, because the DI can pass only one entity manager at DI compile time, while the entity manager used in command is passed at runtime time.

Reply:
Ok then. Let's go with this if there is no other wa. Thanks!